### PR TITLE
HomeWork

### PR DESCRIPTION
**Pull request by DadizhaO**

### CODE REVIEW

**Review by Yanetta — COMMENTED** [review]
Everything is good.

**Review by Yanetta — APPROVED** [review]
Ok!

File: src/main/java/Lesson14/MateGroupServiceImpl.java [hunk 1]
```diff
@@ -63,10 +65,75 @@ public Response updateStudebts(@PathParam("groupId") int groupId, @PathParam("su
...
+    @Path("/humanResources")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addHumanResources(@PathParam("groupId") int groupId, HumanResource humanResource) {
+        MateGroup mateGroup = mateGroups.get(groupId);
+        if (mateGroup != null) {
+            mateGroup.getHumanResources().add(humanResource);
+            return Response.status(Status.ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
+        }
+        return Response.status(404).build();
```
Comment (SerhiiPasko) [inline src/main/java/Lesson14/MateGroupServiceImpl.java]: Bad way. Use special Enums for status-codes. It more readable. It critical. For example another programmer don't know what is it 404

File: src/main/java/Lesson14/MateGroupServiceImpl.java [hunk 2]
```diff
@@ -63,10 +65,75 @@ public Response updateStudebts(@PathParam("groupId") int groupId, @PathParam("su
...
         if (mateGroup != null) {
-            
+
             mateGroup.getStudents().stream().filter(s -> s.getSurname().equals(surname)).forEach(s -> s.setName(name));
             return Response.status(Status.ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
         }
         return Response.status(404).build();
     }
+
+    @PUT
```
Comment (SerhiiPasko) [inline src/main/java/Lesson14/MateGroupServiceImpl.java]: better Post

File: src/main/java/Lesson14/MateGroupServiceImpl.java [hunk 3]
```diff
@@ -63,10 +65,75 @@ public Response updateStudebts(@PathParam("groupId") int groupId, @PathParam("su
...
+    @Path("/humanResources")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addHumanResources(@PathParam("groupId") int groupId, HumanResource humanResource) {
+        MateGroup mateGroup = mateGroups.get(groupId);
+        if (mateGroup != null) {
+            mateGroup.getHumanResources().add(humanResource);
+            return Response.status(Status.ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
```
Comment (SerhiiPasko) [inline src/main/java/Lesson14/MateGroupServiceImpl.java]: better return only code

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

**Review by SerhiiPasko — APPROVED** [review]
Good

**Pull request merged**